source/codegen/proto_helpers.py

```diff
@@ -39,15 +39,30 @@ def get_enum_definitions(enums_to_define, enums):
     enum_definitions.update({enum_name: enum_definition})
   return enum_definitions
 
+
+def is_array_input(parameter: dict):
+  return common_helpers.is_array(parameter["type"]) and common_helpers.is_input_parameter(parameter)
+
+
+def is_decomposable_enum(parameter: dict):
+  """
+  Enums are typically decomposed from a single param into an enum param and an _raw param.
+  The exception is array_inputs which are left as single enum param.
+  This is because protobuf does not support oneof on repeated types, so the standard
```

Comment:
Yeah, you can get around this by making a `message` with the `oneof` field and then use that message as the field type with `repeated` but that gets kind of ugly and more complicated. This sounds good.